model/histogram/float_histogram.go

```diff
@@ -269,21 +266,28 @@ func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
 	h.Sum += other.Sum
 
 	var (
+		hPositiveSpans   = h.PositiveSpans
+		hPositiveBuckets = h.PositiveBuckets
+		hNegativeSpans   = h.NegativeSpans
+		hNegativeBuckets = h.NegativeBuckets
+
 		otherPositiveSpans   = other.PositiveSpans
 		otherPositiveBuckets = other.PositiveBuckets
 		otherNegativeSpans   = other.NegativeSpans
 		otherNegativeBuckets = other.NegativeBuckets
 	)
 
 	if other.Schema < h.Schema {
-		panic(fmt.Errorf("cannot add histogram with schema %d to %d", other.Schema, h.Schema))
+		hPositiveSpans, hPositiveBuckets = reduceResolution(hPositiveSpans, hPositiveBuckets, h.Schema, other.Schema, false, true)
+		hNegativeSpans, hNegativeBuckets = reduceResolution(hNegativeSpans, hNegativeBuckets, h.Schema, other.Schema, false, true)
+		h.Schema = other.Schema
 	} else if other.Schema > h.Schema {
```

Comment:
We have a slight cosmetic preference to write constructs like this as `switch` statements (because then you see the conditions for the two branches at the same indent level) even though this is only one `else if` and not a whole cascade. (The latter would be flagged by the linter.) I know this was there before, but before the first branch was really a "let's just make sure an invariant is true before going on" while now we have really equivalent operations in both branches, so they should also look the same.

Reply:
Sure thing, switched to `switch`.